syntax = "proto2";

package tbclient;

import "tbclient/Abstract.proto";
import "tbclient/ActInfo.proto";
import "tbclient/ActivityInfo.proto";
import "tbclient/AdInfo.proto";
import "tbclient/Agree.proto";
import "tbclient/AlaLiveInfo.proto";
import "tbclient/AnchorInfo.proto";
import "tbclient/AnimationThread.proto";
import "tbclient/AppCode.proto";
import "tbclient/AppInfo.proto";
import "tbclient/BookThread.proto";
import "tbclient/CartoonThread.proto";
import "tbclient/DealInfo.proto";
import "tbclient/DeclareInfo.proto";
import "tbclient/DislikeInfo.proto";
import "tbclient/Guess.proto";
import "tbclient/HotTWThreadInfo.proto";
import "tbclient/JNews.proto";
import "tbclient/Lbs.proto";
import "tbclient/LinkThreadInfo.proto";
import "tbclient/Media.proto";
import "tbclient/MediaNum.proto";
import "tbclient/MultipleForum.proto";
import "tbclient/OriginThreadInfo.proto";
import "tbclient/PbContent.proto";
import "tbclient/PollInfo.proto";
import "tbclient/Post.proto";
import "tbclient/PostList.proto";
import "tbclient/PsInfo.proto";
import "tbclient/PushStatus.proto";
import "tbclient/ReportInfo.proto";
import "tbclient/SkinInfo.proto";
import "tbclient/StarRankIcon.proto";
import "tbclient/TailInfo.proto";
import "tbclient/TaskInfo.proto";
import "tbclient/TbreadDispatch.proto";
import "tbclient/TogetherHi.proto";
import "tbclient/Topic.proto";
import "tbclient/User.proto";
import "tbclient/VideoChannelInfo.proto";
import "tbclient/VideoInfo.proto";
import "tbclient/Voice.proto";
import "tbclient/YulePostActivity.proto";
import "tbclient/Zan.proto";
import "tbclient/ZhiBoInfoTW.proto";

message ThreadInfo {
  repeated Abstract _abstract = 21;
  optional string ab_tag = 108;
  repeated ActInfo act_info = 65;
  optional ActivityInfo activity_info = 132;
  optional Agree agree = 126;
  optional int32 agree_num = 124;
  optional AlaLiveInfo ala_info = 113;
  optional AnchorInfo anchor_info = 55;
  optional AnimationThread animation_info = 99;
  optional AppCode app_code = 89;
  optional AppInfo app_info = 117;
  optional int64 audit_time = 138;
  optional User author = 18;
  optional int64 author_id = 56;
  optional BookThread book_chapter = 102;
  optional CartoonThread cartoon_info = 92;
  optional string category_name = 73;
  optional uint32 cheak_repeat = 107;
  optional string click_url = 32;
  optional string collect_mark_pid = 51;
  optional int32 collect_status = 50;
  optional int32 comment_num = 20;
  optional int32 create_time = 45;
  optional DealInfo deal_info = 98;
  repeated DeclareInfo declare_list = 121;
  repeated DislikeInfo dislike_info = 120;
  optional string ecom = 60;
  repeated TailInfo ext_tails = 90;
  optional int64 fid = 27;
  repeated PbContent first_post_content = 142;
  optional int64 first_post_id = 40;
  optional string fname = 28;
  optional int64 freq_num = 130;
  optional string from = 49;
  optional Guess guess = 63;
  optional int32 has_commented = 48;
  optional TogetherHi high_together = 95;
  optional HotTWThreadInfo hotTWInfo = 71;
  optional int32 hot_weight = 66;
  optional int64 id = 1;
  optional uint32 isLzDeleteAll = 58;
  optional int32 is_activity = 16;
  optional uint32 is_ad = 59;
  optional int32 is_bakan = 12;
  optional uint32 is_book_chapter = 103;
  optional int32 is_bub = 39;
  optional int32 is_called = 136;
  optional uint32 is_copythread = 81;
  optional int32 is_deal = 97;
  optional int32 is_global_top = 42;
  optional int32 is_god = 131;
  optional int32 is_godthread_recommend = 85;
  optional int32 is_good = 10;
  optional int32 is_link_thread = 128;
  optional int32 is_livepost = 30;
  optional int32 is_meizhi = 14;
  optional int32 is_membertop = 54;
  optional int32 is_multiforum_thread = 123;
  optional int32 is_notice = 17;
  optional int32 is_novel = 76;
  optional int32 is_novel_reward = 78;
  optional int32 is_novel_thank = 77;
  optional int32 is_ntitle = 38;
  optional uint32 is_operate_thread = 114;
  optional int32 is_partial_visible = 127;
  optional int32 is_pic = 43;
  optional int32 is_protal = 13;
  optional int32 is_share_thread = 143;
  optional int32 is_story_audit = 134;
  optional int32 is_tbread_dispatch = 115;
  optional int32 is_top = 9;
  optional int32 is_voice_thread = 15;
  optional int32 is_vote = 11;
  optional JNews jid = 75;
  optional uint64 last_read_pid = 106;
  optional User last_replyer = 19;
  optional string last_time = 6;
  optional int32 last_time_int = 7;
  optional string lego_card = 94;
  optional LinkThreadInfo link_info = 129;
  optional string live_post_type = 29;
  optional string livecover_src = 67;
  optional Lbs location = 62;
  repeated Media media = 22;
  repeated MediaNum media_num = 25;
  optional string meizhi_pic = 24;
  optional int32 middle_page_num = 139;
  optional int32 middle_page_pass_flag = 140;
  optional string multi_forum_text = 146;
  repeated MultipleForum multiple_forum_list = 122;
  optional uint32 operator_flag = 82;
  optional OriginThreadInfo origin_thread_info = 141;
  optional Media pic_info = 133;
  optional uint32 pic_num = 84;
  optional string pids = 61;
  optional PollInfo poll_info = 74;
  optional int64 post_id = 52;
  repeated PostList post_list = 44;
  optional int32 post_num = 69;
  optional PsInfo ps_info = 101;
  optional int32 push_end_time = 80;
  optional PushStatus push_status = 91;
  optional string recom_extra = 144;
  optional string recom_reason = 109;
  optional string recom_source = 104;
  optional string recom_weight = 105;
  optional int32 reply_num = 4;
  repeated ReportInfo report_info = 118;
  optional int32 repost_num = 46;
  repeated PbContent rich_abstract = 112;
  repeated PbContent rich_title = 111;
  optional int64 share_num = 135;
  optional int32 show_commented = 31;
  optional SkinInfo skin_info = 100;
  optional StarRankIcon star_rank_icon = 147;
  optional int32 storecount = 68;
  optional TaskInfo task_info = 83;
  optional TbreadDispatch tbread_dispatch_info = 116;
  optional int32 thread_type = 26;
  optional int32 thread_types = 8;
  optional int64 tid = 2;
  optional string tieba_game_information_source = 137;
  optional int32 time = 53;
  optional uint32 timeline = 64;
  optional string title = 3;
  optional Post top_agree_post = 125;
  optional Topic topic = 47;
  optional uint64 trans_num = 145;
  optional ZhiBoInfoTW twzhibo_info = 72;
  optional uint32 valid_post_num = 57;
  optional string video = 33;
  optional AdInfo video_ad_info = 110;
  optional VideoChannelInfo video_channel_info = 119;
  optional string video_cover = 35;
  optional string video_id = 36;
  optional VideoInfo video_info = 79;
  optional string video_mobile_url = 37;
  optional string video_swf = 34;
  optional VideoActive videoactive_info = 96;
  optional int32 view_num = 5;
  repeated Voice voice_info = 23;
  optional YulePostActivity yule_post_activity = 88;
  optional Zan zan = 41;
}

message VideoActive {
  optional string active_name = 2;
  optional string active_url = 3;
  optional string banner_url = 6;
  optional string description = 5;
  optional uint32 id = 1;
  optional string native_url = 4;
  optional string remark = 7;
  repeated ThreadInfo thread_list = 8;
}
